lib/module.js

```diff
@@ -10,117 +10,128 @@ const { promisify } = require('util')
 const { hostname } = require('os')
 
 module.exports = function module (moduleOptions) {
-  const defaults = {
-    path: 'sitemap.xml',
-    hostname: this.options.build.publicPath || undefined,
-    exclude: [],
-    routes: this.options.generate.routes || [],
-    cacheTime: 1000 * 60 * 15,
-    filter: undefined,
-    gzip: false,
-    defaults: {}
+  let sitemaps = []
+  if (Array.isArray(this.options.sitemap)) {
+    sitemaps = this.options.sitemap
+  } else {
+    sitemaps.push(this.options.sitemap)
   }
 
-  const options = Object.assign({}, defaults, this.options.sitemap, moduleOptions)
-  options.pathGzip = options.gzip ? `${options.path}.gz` : options.path
+  checkPathnamesValidity(sitemaps)
+
+  for (let i = 0; i < sitemaps.length; i++) {
+    const defaults = {
```

Comment:
outside the loop

Reply:
line 24:
``` path: sitemaps[i].path || 'sitemap.xml',  ```
has to be inside the loop. Should i change defaults to LET instead of CONST? (declared outside the loop but updated inside the loop?)

Reply:
the default value is overwitten by this method `Object.assign({}, ...)` l34